cmd/samples/fileprocessing/workflow.go

```diff
@@ -66,26 +66,34 @@ func SampleFileProcessingWorkflow(ctx workflow.Context, fileID string) (err erro
 
 func processFile(ctx workflow.Context, fileID string) (err error) {
 	var fInfo *fileInfo
-	err = workflow.ExecuteActivity(ctx, downloadFileActivityName, fileID).Get(ctx, &fInfo)
+	so := &workflow.SessionOptions{
+		CreationTimeout:  time.Minute,
+		ExecutionTimeout: time.Minute,
+	}
+
+	sessionCtx, err := workflow.CreateSession(ctx, so)
+	if err != nil {
+		return err
+	}
+	defer workflow.CompleteSession(sessionCtx)
+
+	err = workflow.ExecuteActivity(sessionCtx, downloadFileActivityName, fileID).Get(sessionCtx, &fInfo)
 	if err != nil {
 		return err
 	}
 
-	// The following activities needs to be run on the same host as the first activity.
-	// This is achieved by scheduling them through this host specific task-list.
-	hCtx := workflow.WithTaskList(ctx, fInfo.HostID)
 	// The short schedule to start timeout is needed to ensure that if host is down an activity doesn't
 	// get stuck in a host specific task list.
 	// Note that this timeout is so short to make sample more vivid. In production applications it
 	// is usually higher.
-	hCtx = workflow.WithScheduleToStartTimeout(hCtx, time.Second*2)
+	sessionCtx = workflow.WithScheduleToStartTimeout(sessionCtx, time.Second*2)
```

Comment:
Is it indeed needed  when using session? I though it is already taken care of.

Reply:
It's possible that an activity is scheduled before the session is regarded as failed (before the session creation activity heartbeat timeout).

Reply:
Then we need to cancel activity explicitly upon session failure.

Reply:
Sure, then each session needs to maintain a list of outstanding activities. I can add that feature.

Reply:
Oh, actually I don't need to do that. I can simply cancel the sessionCtx, then all the activities using that context will be cancelled.

Reply:
Cool this makes a lot of sense. Just make sure to document that behavior of the context.